### PR TITLE
Lordpidey devil fixes

### LINKED ISSUE #21321: Being decapitated as a devil makes it impossible to revive.
Right. So this happened a while ago. I was doing my infernal thing. Going around. Offering deals, harassing the chaplain who wouldn't stop spraying me with holy water. Eventually things got violent and security got involved. I was beaten to a pulp and the Chappie decapitated me with his chain sword. Despite having a lesser infernal form( Lizard person.) and having six souls, I never revived. 

This is kind of a problem maybe. Maybe it's intentional, but info regarding the devil says only being banished or being repeatedly beaten into the ground can get rid of them, so it's a bit deceptive if that was the intended mechanic. The reason I think it's a bug though is I got the message to remain in my body. Stayed in my severed head. Eventually the 'WE RISE AGAIN!' Message appeared but I was not revived. 

 I know Changelings can revive without heads. Could that code be cannibalized for this?

Any other info that might be relevant was I was a TC trade Devil.

### LINKED ISSUE #21326: Runtime error attacking with an empty hand as true devil
runtime error: 
[14:27:45]undefined variable /mob/living/carbon/true_devil/var/martial_art
[14:27:45]proc name: spec attack hand (/datum/species/proc/spec_attack_hand)
[14:27:45]  source file: species.dm,1039
[14:27:45]  usr: Hugyermya, the Lord of all thi... (/mob/living/carbon/true_devil)
[14:27:45]  src: Human (/datum/species/human)
[14:27:45]  usr.loc: the floor (119,134,1) (/turf/open/floor/plasteel)
[14:27:45]  call stack:
[14:27:45]Human (/datum/species/human): spec attack hand(Hugyermya, the Lord of all thi... (/mob/living/carbon/true_devil), Nikolas Rifler (/mob/living/carbon/human), null)
[14:27:45]Nikolas Rifler (/mob/living/carbon/human): attack hand(Hugyermya, the Lord of all thi... (/mob/living/carbon/true_devil))
[14:27:45]Hugyermya, the Lord of all thi... (/mob/living/carbon/true_devil): UnarmedAttack(Nikolas Rifler (/mob/living/carbon/human), 1)
[14:27:45]Hugyermya, the Lord of all thi... (/mob/living/carbon/true_devil): ClickOn(Nikolas Rifler (/mob/living/carbon/human), "icon-x=20;icon-y=6;left=1;scre...")
[14:27:45]Nikolas Rifler (/mob/living/carbon/human): Click(the floor (119,133,1) (/turf/open/floor/plasteel/blue/side), "mapwindow.map", "icon-x=20;icon-y=6;left=1;scre...")

@lordpidey

### LINKED ISSUE #22677: Making people devils doesn't log
In admin log or chat

### PR DESCRIPTION
**Pull request by lordpidey**
Fixes #21326
Fixes #21321
Fixes #22677

Devils with the dancing obligation can no longer break through walls by dropping a dance floor on them

Devils can now be marked as unascendable.  Event devils are unascendable (Event is still admin-only), and devil gamemode devils are ascendable.  If someone is changed via the traitor panel, there's a button for each (and even a button to switch them)

:cl: Lordpidey
fix: Devils can no longer break into areas with sheer force of disco funk
add: The pitchfork of an ascended devil can now break down walls.
add: Hell has decided to at least clothe it's devils when sending them a brand new body.
add: Pitchforks glow red now.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
/:CL: not :/cl:

**Comment by Durkel** [comment]
I may actually start to spawn these now they're not round ending. Interesting at the very least.

**Comment by RemieRichards** [comment]
Sorry pidey, the funk was too good.

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -204,7 +204,8 @@
 	if(M.a_intent == INTENT_DISARM)
 		if(M.buckled_mobs && (src in M.buckled_mobs) && M.riding_datum)
 			M.riding_datum.force_dismount(src)
-	dna.species.spec_attack_hand(M, src)
+	if(istype(M))
```
Comment (AnturK) [inline code/modules/mob/living/carbon/human/human_defense.dm]: I'd prefer if you get rid of the argument bound variables instead and set it in spec_attack_hand body.

Comment (optimumtact) [inline code/modules/mob/living/carbon/human/human_defense.dm]: hellllllllllllllllo

Comment (lordpidey) [inline code/modules/mob/living/carbon/human/human_defense.dm]: I've fixed the problem, but in a different way.  Attack hand should NEVER have had a human as an argument, it should have just been a mob, like the rest of the attack_hand procs.  As such, it's been re-written to the point that this is no longer a concern.

**Comment by lordpidey** [comment]
I just thought of a few more changes that would be helpful, with the new lighting system.  Changes should be ready later today.

**Comment by lordpidey** [comment]
Aright, I've made it so that respawning into a new body doesn't suck AS badly as before.

**Pull request merged**